File: 03. Programming - C#/TestPreparation4Example/Rectangle.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TestPreparation4Example
{
    class Rectangle:Shape
    {
        private double a;
        public double A
        {
            get
            {
                return a;
            }set => a = value;
        }
        private double b;
        public double B
        {
            get
            {
                return b;
            }
            set
            {
                b = value;
            }
        }
        public Rectangle(string name, int numberOfSides,string type, double a, double b)
            :base(name, numberOfSides,type)
        {
            this.a = a;
            this.b = b;
        }
        public override double calculateArea()
        {
            return a * b;
        }

        public override double calculatePerimeter()
        {
            return 2 * (a + b);
        }
    }
}

File: 02. Programming - Java/Lesson7/src/uni/fmi/bachelors/Rectangle.java
package uni.fmi.bachelors;

public class Rectangle extends Shape{

	private double a;
	private double b;
	
	
	public Rectangle(String type, int numberOfSides, int a, int b) {
		super(type, numberOfSides);
		this.a=a;
		this.b=b;
	}

	@Override
	public double calculatePerimeter() {
		return 2*a+2*b;
	}

	@Override
	public double calculateArea() {
		return a*b;
	}

	public double getA() {
		return a;
	}

	public void setA(double a) {
		this.a = a;
	}

	public double getB() {
		return b;
	}

	public void setB(double b) {
		this.b = b;
	}

	
}
